GenderPayGap.WebUI/Search/AutoCompleteSearchService.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using GenderPayGap.Core;
+using GenderPayGap.WebUI.Models.Search;
+using GenderPayGap.WebUI.Search.CachedObjects;
+
+namespace GenderPayGap.WebUI.Search
+{
+
+    internal class AutoCompleteOrganisation
+    {
+        public string EncryptedId { get; set; }
+        public List<AutoCompleteOrganisationName> Names { get; set; }
+        public List<double> Ranks { get; set; }
+    }
+
+    internal class AutoCompleteOrganisationName
+    {
+        public string Name { get; set; }
+        public List<double> Ranks { get; set; }
+    }
+
+
+    public class AutoCompleteSearchService
+    {
+
+        public List<SuggestEmployerResult> Search(string query)
+        {
+            query = query.Trim().ToLower();
+
+            List<string> searchTerms = ExtractSearchTermsFromQuery(query);
+
+            List<SearchCachedOrganisation> allOrganisations = SearchRepository.CachedOrganisations;
+
+            List<SearchCachedOrganisation> matchingOrganisations = GetMatchingOrganisations(allOrganisations, searchTerms);
+
+            List<AutoCompleteOrganisation> orderedOrganisations = CalculateRankings(matchingOrganisations, searchTerms, query);
+
+            List<SuggestEmployerResult> results = ConvertOrganisationsToSearchResults(orderedOrganisations);
+
+            return results;
+        }
+
+        private static List<string> ExtractSearchTermsFromQuery(string query)
+        {
+            return WordSplittingRegex.SplitValueIntoWords(query);
+        }
+
+        private static List<SearchCachedOrganisation> GetMatchingOrganisations(
+            List<SearchCachedOrganisation> allOrganisations,
+            List<string> searchTerms)
+        {
+            return allOrganisations
+                .Where(org => org.Status == OrganisationStatuses.Active || org.Status == OrganisationStatuses.Retired)
+                .Where(org => CurrentOrPreviousOrganisationNameMatchesSearchTerms(org, searchTerms))
+                .ToList();
+        }
+
+        private static bool CurrentOrPreviousOrganisationNameMatchesSearchTerms(SearchCachedOrganisation organisation, List<string> searchTerms)
+        {
+            return organisation.OrganisationNames.Any(on => on.Matches(searchTerms));
+        }
+
+        private static List<AutoCompleteOrganisation> CalculateRankings(
+            List<SearchCachedOrganisation> matchingOrganisations,
+            List<string> searchTerms,
+            string query)
+        {
+            return matchingOrganisations
+                .Select(organisation => CalculateRankForOrganisation(organisation, searchTerms, query))
+                .RankHelperOrderByListOfDoubles(name => name.Ranks)
+                .ThenBy(mar => mar.Names[0].Name)
+                .Take(10)

```

Comment:
From the method name it is also not clear that it was going to truncate that list.